plugins/linux-v4l2/v4l2-output.c

```diff
@@ -111,51 +121,42 @@ static void *virtualcam_create(obs_data_t *settings, obs_output_t *output)
 	return vcam;
 }
 
-static bool try_connect(void *data, const char *device)
+static bool try_configure(void *data, int device)
 {
 	struct virtualcam_data *vcam = (struct virtualcam_data *)data;
-	struct v4l2_format format;
-	struct v4l2_capability capability;
-	struct v4l2_streamparm parm;
-
-	uint32_t width = obs_output_get_width(vcam->output);
-	uint32_t height = obs_output_get_height(vcam->output);
-
-	vcam->frame_size = width * height * 2;
-
-	vcam->device = open(device, O_RDWR);
-
-	if (vcam->device < 0)
-		return false;
-
-	if (ioctl(vcam->device, VIDIOC_QUERYCAP, &capability) < 0)
-		goto fail_close_device;
+	vcam->device = device;
 
+	struct v4l2_format format = {0};
 	format.type = V4L2_BUF_TYPE_VIDEO_OUTPUT;
 
 	if (ioctl(vcam->device, VIDIOC_G_FMT, &format) < 0)
-		goto fail_close_device;
+		return false;
 
 	struct obs_video_info ovi;
 	obs_get_video_info(&ovi);
 
-	memset(&parm, 0, sizeof(parm));
+	struct v4l2_streamparm parm = {0};
 	parm.type = V4L2_BUF_TYPE_VIDEO_OUTPUT;
 
 	parm.parm.output.capability = V4L2_CAP_TIMEPERFRAME;
 	parm.parm.output.timeperframe.numerator = ovi.fps_den;
 	parm.parm.output.timeperframe.denominator = ovi.fps_num;
 
 	if (ioctl(vcam->device, VIDIOC_S_PARM, &parm) < 0)
-		goto fail_close_device;
+		return false;
+
+	uint32_t width = obs_output_get_width(vcam->output);
+	uint32_t height = obs_output_get_height(vcam->output);
+	vcam->frame_size = width * height * 2;
 
 	format.fmt.pix.width = width;
 	format.fmt.pix.height = height;
+	format.fmt.pix.field = V4L2_FIELD_NONE;
```

Comment:
what is the purpose of this change?

Reply:
tbh I am not sure if this makes a difference in any specific scenarios, but the default is V4L2_FIELD_ANY (0) whereas V4L2_FIELD_NONE (1) is for progressive formats (not interlaced). 
Setting this value to hopefully prevent potential odd behavior.
  
https://linuxtv.org/downloads/legacy/video4linux/API/V4L2_API/spec/ch03s06.html